sites/public/src/lib/helpers.tsx

```diff
@@ -202,7 +202,9 @@ export const untranslateMultiselectQuestion = (
 
           if (option.extraData) {
             option.extraData.forEach((extra) => {
-              extra.key = selectedOption.untranslatedText ?? selectedOption.text
+              if (!extra.key) {
+                extra.key = selectedOption.untranslatedText ?? selectedOption.text
+              }
```

Comment:
that one was overwriting key's.
Tbh not sure what's the idea behind, but i assumed that overwriting it wasn't the goal here, it was more like add missing key.

Reply:
This change was added here because if you submit with a different language we want to make sure the option key is in English (untranslatedText). But I don't think the key of the extraData needs to be updated. I like what you did here and we can look later to see if this forEach is needed at all